objects/src/mock/chain.rs

```diff
@@ -55,3 +55,247 @@ pub fn mock_chain_data(consumed_notes: &mut [Note]) -> ChainMmr {
+        objects: Objects<R>,
+
+        // Objects that have been created and are waiting for a block.
+        //
+        // These objects will become available once the block is sealed.
+        #[cfg_attr(feature = "serde", serde(skip))]
+        pending_objects: Objects<R>,
+    }
+
+    impl<R: Rng + SeedableRng> MockChain<R> {
+        pub fn new(mut rng: R) -> Result<Self, rand::Error> {
+            let account_rng = R::from_rng(&mut rng).expect("rng seeding failed");
+            let account_id_builder = AccountIdBuilder::new(account_rng);
+            Ok(Self {
+                chain: ChainMmr::default(),
+                data: DefaultMerkleStore::default(),
+                blocks: vec![],
+                nullifier_db: TieredSmt::default(),
+                rng,
+                account_id_builder,
+                objects: Objects::new(),
+                pending_objects: Objects::new(),
+            })
+        }
+
+        /// Creates the next block.
+        ///
+        /// This will also make all the objects currently pending available for use.
+        pub fn seal_block(&mut self) -> BlockHeader {
+            // Update the DBs
+            // TODO: self.nullifier
+            // TODO: accounts
+            // TODO: notes
+
+            let previous = self.blocks.last();
+            let block_num: u64 = self.blocks.len().try_into().expect("usize to u64 failed");
+            let block_num: Felt = block_num.into();
+            let peaks = self.chain.mmr().accumulator();
+            let chain_root: Digest = peaks.hash_peaks().into();
+            let prev_hash = previous.map_or(Digest::default(), |header| header.hash());
+            let nullifier_root = self.nullifier_db.root();
+            let version = Felt::ZERO;
+            let timestamp =
+                previous.map_or(TIMESTAMP_START, |header| header.timestamp() + TIMESTAMP_STEP);
+
+            // TODO: Set batch_root and proof_hash to the correct values once the kernel is
+            // available.
+            let batch_root = Digest::default();
+            let proof_hash = Digest::default();
+
+            let new = BlockHeader::new(
+                prev_hash,
+                block_num,
+                chain_root,
+                Default::default(),
+                nullifier_root,
+                Default::default(),
+                batch_root,
+                proof_hash,
+                version,
+                timestamp,
+            );
+
+            self.blocks.push(new);
+            self.chain.mmr_mut().add(new.hash());
+            self.objects.extend(&mut self.pending_objects);
+
+            new
+        }
+
+        /// Creates a [Account] and add to the list of pending objects.
+        pub fn new_account<C, S, A>(
+            &mut self,
+            code: C,
+            storage: S,
+            assets: A,
+            immutable: bool,
+            on_chain: bool,
+        ) -> Result<AccountId, AccountError>
```

Comment:
It is a bit unfortunate that we have to pass all of these arguments, since the builder has internal state for it already. But if we want to configure the type of account created this seems the best option.

Reply:
Another option (or rather an addition to this method) is to have some number of more specialized account constructors - e.g., `new_basic_wallet`.

Reply:
added